controlpanel/api/aws.py

```diff
@@ -221,32 +224,31 @@ class S3AccessPolicy:
 
     def __init__(self, policy):
         self.policy = policy
-        self._policy_document = None
         self.statements = {}
 
-        self._load()
-
-    def _load(self):
         try:
-            # ensure version is set
-            self.policy_document['Version'] = '2012-10-17'
+            policy_document = self.load_policy_document()
 
         except self.policy.meta.client.exceptions.NoSuchEntityException:
             # create an empty s3 access policy
-            self.put(policy_document=deepcopy(BASE_S3_ACCESS_POLICY))
+            policy_document = deepcopy(BASE_S3_ACCESS_POLICY)
+            self.put(policy_document=policy_document)
+
+        # ensure version is set
+        policy_document['Version'] = '2012-10-17'
```

Comment:
If we find that the logic to initialise the `self.policy_document` attribute is getting a bit hairy/low-level we may consider moving this logic into a "private" helper method. Maybe even in  `load_policy_document()`.

If you feel like assigning `self.policy_document` from within `load_policy_document` is a bit sneaky you may have something like:

```python
self.policy_document = self.get_policy_document()
```

Reply:
No, the `load_policy_document` method is deliberately as concise as possible to cover only the code that needs to be overridden in the subclass.